packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -354,23 +369,39 @@ export class QueuedRequestController extends BaseController<
     request: QueuedRequestMiddlewareJsonRpcRequest,
     requestNext: () => Promise<void>,
   ): Promise<void> {
+    if (request.networkClientId === undefined) {
+      // This error will occur if selectedNetworkMiddleware does not precede queuedRequestMiddleware in the middleware stack
+      throw new Error(
+        'Error while attempting to enqueue request: networkClientId is required.',
+      );
+    }
     if (this.#originOfCurrentBatch === undefined) {
       this.#originOfCurrentBatch = request.origin;
     }
+    if (this.#networkClientIdOfCurrentBatch === undefined) {
+      this.#networkClientIdOfCurrentBatch = request.networkClientId;
+    }
 
+    const isMultichainRequestToQueue =
+      this.#originOfCurrentBatch === request.origin &&
+      this.#networkClientIdOfCurrentBatch !== request.networkClientId;
     try {
       // Queue request for later processing
       // Network switch is handled when this batch is processed
       if (
         this.state.queuedRequestCount > 0 ||
-        this.#originOfCurrentBatch !== request.origin
+        this.#originOfCurrentBatch !== request.origin ||
+        isMultichainRequestToQueue
```

Comment:
I think this can be simplified to: ?
```suggestion
         this.#networkClientIdOfCurrentBatch !== request.networkClientId
```

Reply:
yes!

Reply:
done here: https://github.com/MetaMask/core/pull/4718/commits/2603f2530bddf9b1301d37cad319fdd93e5b8a28